driver-core/src/main/com/mongodb/client/model/CreateCollectionOptions.java

```diff
@@ -194,6 +199,63 @@ public CreateCollectionOptions collation(@Nullable final Collation collation) {
         return this;
     }
 
+    /**
+     * Returns the expire-after option.
+     *
+     * @return the expire-after option, which may be null.
+     * @since 4.3
+     * @mongodb.driver.manual core/timeseries-collections/ Time-series collections
+     */
+    @Nullable
+    public Duration getExpireAfter() {
+        return expireAfter;
+    }
+
+    /**
+     * Sets the expire-after option.
+     *
+     * <p>
+     * A duration indicating after how long old time-series data should be deleted. Partial seconds are ignored.
+     * </p>
+     * <p>
+     * Currently applies only to time-series collections, so if this value is set then so must the time-series options
+     * </p>
+     * @param expireAfter the expire-after duration
+     * @return this
+     * @since 4.3
+     * @see #timeSeriesOptions(TimeSeriesOptions)
+     * @mongodb.driver.manual core/timeseries-collections/ Time-series collections
+     */
+    public CreateCollectionOptions expireAfter(@Nullable final Duration expireAfter) {
```

Comment:
Prior to 4.0 past we might have used this signature:

    expireAfter(long expireAfter, TimeUnit timeUnit)

I think `Duration` is better, and now we can use it (since it was introduced in Java 8), but there are downsides

1. Inconsistent with existing API, which uses `TimeUnit`
2. We are dropping the nanoseconds portion of `Duration` (though we could it?)

Also, since the actual field is `expireAfterSeconds`, we could instead do:

    expireAfterSeconds(long seconds)

but that seems unfriendly.

Reply:
Using `Duration` is great. I wish we decided to start using `Optional` in public APIs.

Reply:
Also why isn't this part of `TimeSeriesOptions`?

Reply:
The server specification indicates that it's separate.  There is some thought that expireAfter could apply to other types of collections besides time-series in the future.  Jeremy left a [comment](https://jira.mongodb.org/browse/DRIVERS-1438?focusedCommentId=3854462&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-3854462) in the associated DRIVERS ticket about it

Reply:
I'm sympathetic to the view that we shouldn't use `Duration` in just one place and am fine using `TimeUnit` instead.  @stIncMale would you object to this decision?

Reply:
While I understand the API uniformity sentiment<sup>1</sup>, its disadvantage that the global API refactoring may either never happen, or (the best case scenario), may happen in a distant future, thus depriving the API from the benefits of better approaches for longer than it would have been otherwise due to hard restrictions (neither `Duration` nor `Optional` could have been used until the driver required the Java SE 8 at runtime). That being said, I will greet new APIs that use `Optional`/`java.time`, but I will also **not object** to new APIs that use the old approaches.

----

<sup>1</sup> It is especially relevant when mixing `@Nullable` and `Optional`, because observing `Optional` in some places may give a false expectation that all non-`Optional` return values are not nullable. This is likely one of the the reasons the Java SE API adopted neither `Optional` nor the new `java.time` API.

Reply:
I opted for `long` and `TimeUnit` to match how we did `maxTimeMS`.

Reply:
PTAL, @rozza

Reply:
Added JAVA-4191 to track improving the whole API.